libpromises/eval_context.c

```diff
@@ -1384,6 +1384,27 @@ Promise *EvalContextStackPushPromiseIterationFrame(EvalContext *ctx, size_t iter
 
     EvalContextStackPushFrame(ctx, StackFrameNewPromiseIteration(pexp, iter_ctx, iteration_index));
 
+    for (size_t i = 0; i < SeqLength(pexp->conlist); i++)
+    {
```

Comment:
This for loop make me skeptical, it's a hot path, and it will cause a slow down for sure. Still thinking of a better approach.